### PR TITLE
Allow Gutenberg core blocks

### PR DESCRIPTION
**Pull request by karabishopgreenpeace**
# Pull Request Overview
Allow native Gutenberg blocks using code supplied by Japan.  See https://www.notion.so/greenpeaceaustraliapacific/P4-Allow-Gutenberg-Native-Blocks-8f6d40ac30a442a7b023b57f79cebe7e?pvs=4

# 📷 Screenshots

image

## 🤨 Any Issues or Questions?

We will add to this list as we find the names of other blocks.

## 🏁 PR Checklist

Reviewer checklist:
- [ ] ✨ This PR includes new user-facing feature and/or settings
- [ ] 📝 Documentation for new features/settings are in PR description (to go to wiki or Coral docs) 

Contributor checklist:
- [x] Reference a GitHub issue, Jira ticket, or Notion card in the description 💯
- [ ] 📝 Update the CHANGELOG.md (Put under `[Unreleased]`). Include link to this PR.
- [ ] 📝 Add JSDoc code documentation
- [ ] 🏷 Add flow types. (If needed.)
- [ ] ✅ Add unit tests (`*.test.js`) and/or Cypress Tests.
- [ ] 📖 Create Storybook stories. (For any applicable Components.)
- [ ] 📸 Update snapshots.  Run `yarn test:watch`, press 'a' for "all tests", and "u" to update snaps if needed.
- [ ] Run tests (`yarn test` and `yarn flow`) and clean up any errors 🚀
- [x] Select 1-2 Reviewers
- [x] Feel good about yourself, you're doing great! 🥳

### CODE REVIEW

File: functions.php [hunk 1]
```diff
@@ -43,3 +43,20 @@ function get_template_content($name = null){
 }
 
 add_action('wp_enqueue_scripts', 'register_custom_script');
+
+// Allow Gutenberg core blocks
+function p4_child_theme_gpap_whitelist_blocks( $allowed_blocks, $post ) {
+	$allowed = is_array($allowed_blocks) ? $allowed_blocks : array();
+  array_push($allowed, 'core/cover');
+	array_push($allowed, 'core/post-title');
```
Comment (techieshark) [inline functions.php:51]: consistent whitespace:

```suggestion
  array_push($allowed, 'core/cover');
  array_push($allowed, 'core/post-title');
```

File: functions.php [hunk 2]
```diff
@@ -43,3 +43,20 @@ function get_template_content($name = null){
 }
 
 add_action('wp_enqueue_scripts', 'register_custom_script');
+
+// Allow Gutenberg core blocks
+function p4_child_theme_gpap_whitelist_blocks( $allowed_blocks, $post ) {
```
Comment (techieshark) [inline functions.php:48]: (nit) better name
```suggestion
function p4_child_theme_gpap_add_allowed_blocks( $allowed_blocks, $post ) {
```

File: functions.php [hunk 3]
```diff
@@ -43,3 +43,20 @@ function get_template_content($name = null){
...
+  array_push($allowed, 'core/post-featured-image');
+  array_push($allowed, 'core/post-content');
+  array_push($allowed, 'core/post-author');
+  array_push($allowed, 'core/post-date');
+  array_push($allowed, 'core/post-modified-date');
+  array_push($allowed, 'core/post-categories');
+  array_push($allowed, 'core/post-tags');
+	return $allowed;
+}
+add_filter('allowed_block_types', 'p4_child_theme_gpap_whitelist_blocks', 11, 2);
```
Comment (techieshark) [inline functions.php:62]: better name 

```suggestion
add_filter('allowed_block_types', 'p4_child_theme_gpap_add_allowed_blocks', 11, 2);
```

see https://www.vice.com/en/article/we-need-to-stop-saying-blacklist-and-whitelist/

**Review by techieshark — CHANGES_REQUESTED** [review]
some easy changes. After these changes unless Ashley jumps in to object, I'd say merge this and test it on the P4 dev site, and if @ashleytowner sees it works there, they can make a release to push to prod. ... but only if it takes less than say 20 minutes, because Kylie says highest priority is the Action Message Endpoint work (aka "CAPI" as you may know it by).

**Review by ashleytowner — APPROVED** [review]
LGTM

File: functions.php [hunk 4]
```diff
@@ -43,3 +43,20 @@ function get_template_content($name = null){
...
```
Comment (techieshark) [inline functions.php:62]: needs to be renamed @ashleytowner @karabishopgreenpeace

**Review by techieshark — APPROVED** [review]
LGTM

**Review by ashleytowner — APPROVED** [review]

**Pull request merged**